parachain/pallets/verifier/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+#![allow(unused_variables)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_system as system;
+use frame_support::{decl_module, decl_storage, decl_event, decl_error,
+	dispatch::DispatchResult};
+use sp_runtime::traits::Hash;
+use sp_std::prelude::*;
+
+use artemis_core::{AppId, Message, Verifier, VerificationInput};
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub trait Trait: system::Trait {
+	type Event: From<Event> + Into<<Self as system::Trait>::Event>;
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as VerifierModule {
+		RelayKey get(fn key) config(): T::AccountId;
+		pub VerifiedPayloads: map hasher(blake2_128_concat) T::Hash => ();
```

Comment:
Just as a note for the future to keep the storage size bounded: 

We could simply have a double map here, where the first key is the block height of the received message. We can the define a `maxTailLength`, e. g. 200 blocks, and just remove all storage keys where `height < currentHeight - 200`. At the same time, we decline verification for older blocks.

Reply:
To support this, we could already start storing the double map here, but not do the tail length check and failures?

Reply:
Added #83 to track.